rclcpp/include/rclcpp/memory_strategy.hpp

```diff
@@ -101,6 +155,11 @@ class MemoryStrategy
   std::vector<rclcpp::subscription::SubscriptionBase::SharedPtr> subs;
   std::vector<rclcpp::service::ServiceBase::SharedPtr> services;
   std::vector<rclcpp::client::ClientBase::SharedPtr> clients;
+
+  std::vector<void *> subscription_handles;
```

Comment:
I think it's fine for the vectors to be initialized to empty. Specifying a number of handles reserve beforehand seems unnecessarily complicated for the default case. The alternative MemoryStrategy implementations take max numbers of handles as their arguments, however.

When `borrow_handles` is called, if the vector size is smaller than the requested size `n`, then the vector will call `resize(n)`.

http://en.cppreference.com/w/cpp/container/vector/resize

My interpretation is that calling resize for a size greater than the vector's current size will insert elements into the vector to make it grow to the required size.